### PR TITLE
Execute `a.out` after successful linking of the executable

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Fixes: #1130

### CODE REVIEW

File: src/bin/lpython.cpp [hunk 1]
```diff
@@ -817,6 +817,13 @@ int link_executable(const std::vector<std::string> &infiles,
                 std::cout << "The command '" + cmd + "' failed." << std::endl;
                 return 10;
             }
+            if (outfile == "a.out") {
+                err = system("./a.out");
+                if (err) {
+                    std::cout << "The command './a.out' failed." << std::endl;
+                    return 10;
```
Comment (certik) [inline src/bin/lpython.cpp:824]: Does this work on Windows as well?

Comment (Thirumalai-Shaktivel) [reply src/bin/lpython.cpp:824]: This doesn't work on Windows. I wonder why?

Comment (Thirumalai-Shaktivel) [reply src/bin/lpython.cpp:824]: Updated, Now it works on windows as well.

**Comment by Thirumalai-Shaktivel** [comment]
What is the command used to switch the backend?
Following doesn't work on windows:
```py
$ src\bin\lpython.exe --backend cpp examples\expr2.py
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
LCompilersException: Unsupported backend.
```

**Comment by Thirumalai-Shaktivel** [comment]
This might be reason: 
https://github.com/lcompilers/lpython/blob/7a07cb98a35cce504b3a5eba4ec6efa9b18a6ab3/src/bin/lpython.cpp#L1375-L1384

**Review by certik — APPROVED** [review]
I think this is fine, this will give the expected outcome.

I don't know yet if we need if we want to even generate `a.out` or not, but this PR is iterative, so we used to generate it and we still do, we just run it as well.

**Comment by certik** [comment]
Feel free to merge when ready.

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the review.
Yup, this is ready. Let's merge it.

**Pull request merged**